### PR TITLE
Cookie notification

### PR DESCRIPTION
**Pull request by davidpmccormick**
## What is this PR trying to achieve?

Adding a cookie notification banner to the bottom of the screen. Clicking the close button will drop a cookie on the user's browser that expires long in the future (2036, before the [JavaScript year 2038 cookie bug](https://bugzilla.mozilla.org/show_bug.cgi?id=27070)) and removes the banner on any future page loads.

## What does it look like?

Desktop:
screen shot 2016-12-23 at 12 26 07

Mobile:
screen shot 2016-12-23 at 12 26 30

## Have the following been considered/are they needed?

- [x] A11y testing - if this PR affects a user interface component, have you run pa11y against the component and fixed/verified all errors, warnings and notices?

### CODE REVIEW

File: client/js/components/cookie-notification.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import cookie from 'cookie-cutter';
+const COOKIE_NAME = 'cookies_accepted';
```
Comment (jamesgorrie) [inline client/js/components/cookie-notification.js]: Is there reason to capitalise this?

File: client/js/components/cookie-notification.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+const cookieNotification = (el) => {
+  const isAccepted = cookie.get(COOKIE_NAME);
+  const closeButton = document.getElementById('cookie-notification-close');
+
+  if (isAccepted) {
+    el.parentNode.removeChild(el);
+
+    return;
+  }
```
Comment (jamesgorrie) [inline client/js/components/cookie-notification.js]: Wrapping this in an else might be more clear?

File: client/js/util.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ const KEYS = {
   TAB: 9,
   ENTER: 13,
   ESCAPE: 27,
+  SPACEBAR: 32,
```
Comment (jamesgorrie) [inline client/js/util.js]: Where's this used?

File: client/js/components/cookie-notification.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import cookie from 'cookie-cutter';
+const COOKIE_NAME = 'cookies_accepted';
+
+const cookieNotification = (el) => {
+  const isAccepted = cookie.get(COOKIE_NAME);
+  const closeButton = document.getElementById('cookie-notification-close');
+
+  if (isAccepted) {
+    el.parentNode.removeChild(el);
```
Comment (jamesgorrie) [inline client/js/components/cookie-notification.js]: We could just not add the visible class?
e.g.
```JS
if (!isAccepted) {
    el.classList.add('cookie-notification--is-visible');
}
```

File: client/js/components/cookie-notification.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    return;
+  }
+
+  el.classList.add('cookie-notification--is-visible');
+
+  const setCookie = () => {
```
Comment (jamesgorrie) [inline client/js/components/cookie-notification.js]: Why not have this as a function outside of the init, or just attach it anonymously to the event handler?

File: client/scss/components/_cookie_notification.scss [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  .icon__shape {
+    fill: color('white');
+  }
+
+  &.cookie-notification--is-visible {
+    display: block;
+  }
+
+  &.cookie-notification--is-faded {
+    opacity: 0;
```
Comment (gestchild) [inline client/scss/components/_cookie_notification.scss]: Just changing the opacity might cause issues, it'll prevent clicking/selecting anything that is under the message, which might happen when it's not visible.

File: client/js/util.js (hunk 3 above)
Comment (davidpmccormick) [inline client/js/util.js]: Ah thanks. Was a hangover from before I was using button and was replicating the spacebar keydown event. Will remove.

File: client/scss/components/_cookie_notification.scss (hunk 6 above)
Comment (davidpmccormick) [inline client/scss/components/_cookie_notification.scss]: Good point. I'll change the z-index after a transition-delay so that the fade is visible.

**Pull request merged**